ui/imports/shared/popups/keycard/KeycardPopupContent.qml

```diff
@@ -0,0 +1,282 @@
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.wrongPin ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.wrongKeychainPin ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.changingKeycardPinSuccess ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.changingKeycardPinFailure ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.pinSet ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.pinVerified)
+            {
+                return keycardPinComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.createPuk ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.repeatPuk ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.enterPuk ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.wrongPuk)
+            {
+                return keycardPukComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.enterSeedPhrase ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.wrongSeedPhrase)
+            {
+                return enterSeedPhraseComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.seedPhraseDisplay)
+            {
+                return seedPhraseComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.seedPhraseEnterWords)
+            {
+                return enterSeedPhraseWordsComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.enterPassword ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.enterBiometricsPassword ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.wrongBiometricsPassword ||
+                    root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.wrongPassword) {
+                return passwordComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.enterKeycardName) {
+                return enterNameComponent
+            }
+            if (root.sharedKeycardModule.currentState.stateType === Constants.keycardSharedState.createPairingCode) {
+                return enterPairingCodeComponent
+            }
+
+            return undefined
```

Comment:
why not `switch`?

Reply:
@borismelnik yes, thanks for pointing that, I updated all conditions... a reason why we had complex `if` conditions there is when I started on that didn't know how big and complex that part will be later and for first flow I used if, then continued with it, but now it's definitely time to change that.